### PR TITLE
createRouter accepts array of route arrays

### PR DESCRIPTION
**Pull request by stackoverfloweth**
adds overload to createRouter that supports passing an array of route arrays

```ts
const aRoutes = createRoutes([
  { name: 'a-route-1', path: '/a-route-1', component },
  { name: 'a-route-2', path: '/a-route-2', component },
])
const bRoutes = createRoutes([
  { name: 'b-route-1', path: '/b-route-1', component },
  { name: 'b-route-2', path: '/b-route-2', component },
])

createRouter([aRoutes, bRoutes])
```

This will be extremely useful for devs now that they have access to `createExternalRoutes` as well as `createRoutes`.

### CODE REVIEW

File: src/services/createRouter.ts [hunk 1]
```diff
@@ -47,7 +48,10 @@ type RouterUpdateOptions = {
  * const router = createRouter(routes)
  * ```
  */
-export function createRouter<const T extends Routes>(routes: T, options: RouterOptions = {}): Router<T> {
+export function createRouter<const T extends Routes>(routes: T, options?: RouterOptions): Router<T>
+export function createRouter<const T extends Routes>(arrayOfRoutes: T[], options?: RouterOptions): Router<T>
```
Comment (pleek91) [inline src/services/createRouter.ts:52]: I'm surprised we don't need to flatten this. But I suppose T is a union so we get a union of routes on the router?

File: src/utilities/guards.ts [hunk 2]
```diff
@@ -19,3 +19,7 @@ export function hasProperty<
 
   return true
 }
+
+export function isNestedArray<T extends any[] | readonly any[]>(value: T | T[]): value is T[] {
+  return value.length > 0 && Array.isArray(value[0])
```
Comment (pleek91) [inline src/utilities/guards.ts:24]: Should this maybe be `value.every(item => Array.isArray(item))`

Comment (stackoverfloweth) [reply src/utilities/guards.ts:24]: I suppose that's perhaps technically more accurate given that this is a utility, but in the context of createRouter, if the 1st item in the array is an array, they are all going to be. I'd prefer not to iterate through potentially a large set to prove what Typescript types should have already enforced?

File: src/services/createRouter.ts (hunk 1 above)
Comment (stackoverfloweth) [reply src/services/createRouter.ts:52]: I agree, I was pleasantly surprised but it definitely does get correctly merged types

image

File: src/utilities/guards.ts (hunk 2 above)
Comment (pleek91) [reply src/utilities/guards.ts:24]: I don't feel strongly. But it's probably a small array and it reads better IMO. Merge away as is 👍

File: src/services/createRouter.ts (hunk 1 above)
Comment (pleek91) [reply src/services/createRouter.ts:52]: Nice, simpler than I imagined.

**Review by pleek91 — APPROVED** [review]

**Pull request merged**